tests/e2e/volumecreate_test.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/vmware/docker-volume-vsphere/tests/utils/govc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+
+	. "gopkg.in/check.v1"
+)
+
+const ErrorVolumeCreate = "Error response from daemon: create"
+
+type VolumeCreateTestSuite struct {
+	hostIP     string
+	esxIP      string
+	dsNameList []string
+	volumeList []string
+}
+
+func (s *VolumeCreateTestSuite) SetUpSuite(c *C) {
+	s.hostIP = os.Getenv("VM2")
+	s.esxIP = os.Getenv("ESX")
+	s.dsNameList = govc.GetDatastoreList()
+	log.Printf("Datastores found are %v", s.dsNameList)
+}
+
+func (s *VolumeCreateTestSuite) SetUpTest(c *C) {
+	// clean the list of volumes created
+	s.volumeList = s.volumeList[:0]
+}
+
+func (s *VolumeCreateTestSuite) TearDownTest(c *C) {
+	volList := ""
+	for _, name := range s.volumeList {
+		volList = volList + " " + name
+	}
+	if volList != "" {
+		out, err := dockercli.DeleteVolume(s.hostIP, volList)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+}
+
+var _ = Suite(&VolumeCreateTestSuite{})
+
+// Valid volume names test
+// 1. having 100 chars
+// 2. having various chars including alphanumerics
+// 3. ending in 5Ns
+// 4. ending in 7Ns
+// 5. contains @datastore (valid name)
+// 6. contains multiple '@'
+// 7. contains unicode character
+func (s *VolumeCreateTestSuite) TestValidName(c *C) {
+	log.Printf("START: volumecreate_test.TestValidName")
+
+	s.volumeList = append(s.volumeList, inputparams.GetVolumeNameOfSize(100))
+	s.volumeList = append(s.volumeList, "Volume-0000000-****-###")
+	s.volumeList = append(s.volumeList, "Volume-00000")
+	s.volumeList = append(s.volumeList, "Volume-0000000")
+	s.volumeList = append(s.volumeList, inputparams.GetVolumeNameWithTimeStamp("abc")+"@"+s.dsNameList[0])
+	s.volumeList = append(s.volumeList, inputparams.GetVolumeNameWithTimeStamp("abc")+"@@@@"+s.dsNameList[0])
+	s.volumeList = append(s.volumeList, inputparams.GetVolumeNameWithTimeStamp("Volume-你"))
+
+	for _, name := range s.volumeList {
+		out, err := dockercli.CreateVolume(s.hostIP, name)
+		c.Assert(err, IsNil, Commentf(out))
+
+		isAvailable := verification.CheckVolumeAvailability(s.hostIP, name)
+		c.Assert(isAvailable, Equals, true, Commentf("Volume %s is not available after creation", name))
+	}
+
+	log.Printf("END: volumecreate_test.TestValidName")
+}
+
+// Invalid volume names test
+// 1. having more than 100 chars
+// 2. ending -NNNNNN (6Ns)
+// 3. contains @invalid datastore name
+func (s *VolumeCreateTestSuite) TestInvalidName(c *C) {
+	log.Printf("START: volumecreate_test.TestInvalidName")
+
+	var invalidVolList []string
+
+	invalidVolList = append(invalidVolList, inputparams.GetVolumeNameOfSize(101))
+	invalidVolList = append(invalidVolList, "Volume-000000")
+	invalidVolList = append(invalidVolList, inputparams.GetVolumeNameWithTimeStamp("Volume")+"@invalidDatastore")
+
+	for _, name := range invalidVolList {
+		out, _ := dockercli.CreateVolume(s.hostIP, name)
```

Comment:
Refactored this code. err is not nil and output message contains create error substring